staging/src/k8s.io/component-base/metrics/wrappers.go

```diff
@@ -65,6 +68,70 @@ type GaugeMetric interface {
 	SetToCurrentTime()
 }
 
+// GaugeMetricVec is a collection of Gauges that differ only in label values.
+// This is really just one Metric.
+// It might be better called GaugeVecMetric, but that pattern of name is already
+// taken by the other pattern --- which is treacherous.  The treachery is that
+// WithLabelValues can return an object that is permanently broken (i.e., a noop).
+type GaugeMetricVec interface {
+	Set(value float64, labelValues ...string)
+	Inc(labelValues ...string)
+	Dec(labelValues ...string)
+	Add(delta float64, labelValues ...string)
+	SetToCurrentTime(labelValues ...string)
+
+	SetForLabels(value float64, labels map[string]string)
```

Comment:
Do we really need those methods? I don't see them implemented by e.g. HistogramVec, and so I don't see a reason why we really would need them here. I really prefer consistency...